### PR TITLE
updating module overrides

### PR DESCRIPTION
**Pull request by ezolenko**
## Summary

Fix for #437

## Details

- keeping node* `moduleResolution` option as is ("node", "node16", "nodenext")
- overriding "classic" and empty to Node10 (old behavior was overriding everything to Node10)
- overriding "bundler" to Node16

### CODE REVIEW

File: dist/parse-tsconfig.d.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import { RollupContext } from "./context";
-import { IOptions } from "./ioptions";
-export declare function parseTsConfig(context: RollupContext, pluginOptions: IOptions): {
-    parsedTsConfig: import("typescript").ParsedCommandLine;
-    fileName: string | undefined;
-};
```
Comment (agilgur5) [inline dist/parse-tsconfig.d.ts:6]: these look to be exactly the same, a little confused why it's showing as a diff? 🤔  
looks to be the same spacing too?

seems to have happened for several of the generated declarations

File: src/get-options-overrides.ts [hunk 2]
```diff
@@ -16,13 +16,28 @@ export function getOptionsOverrides({ useTsconfigDeclarationDir, cacheRoot }: IO
...
+		case undefined:
+			overrides.moduleResolution = tsModule.ModuleResolutionKind.Node10;
+			break;
+		case tsModule.ModuleResolutionKind.Bundler:
+		default:
+			overrides.moduleResolution = tsModule.ModuleResolutionKind.Node16;
+	}
+
 	if (preParsedTsconfig.options.module === undefined)
 		overrides.module = tsModule.ModuleKind.ES2015;
```
Comment (agilgur5) [inline src/get-options-overrides.ts:42]: I believe this line will never get hit now / is dead code? test coverage should confirm that

**Comment by agilgur5** [comment]
Replaced by #453

**Comment by agilgur5** [comment]
Or rather, this got split into multiple PRs it looks like:
- #453 for `moduleResolution` changes
- #451 for `cts` + `mts`
- #450 for `module: 'ES2022'`

Much nicer and easier to read as separate PRs!

File: dist/parse-tsconfig.d.ts (hunk 1 above)
Comment (agilgur5) [reply dist/parse-tsconfig.d.ts:6]: self resolved as this PR was closed and this change doesn't exist in the other PRs

File: src/get-options-overrides.ts (hunk 2 above)
Comment (agilgur5) [reply src/get-options-overrides.ts:42]: oh I misread this, my bad. `module` vs. `moduleResolution` above

**Comment by ezolenko** [comment]
Yeah, ignore that PR, I started too many changes at once and updated deps too far and that required yet more changes and so on (we'd still have to figure out ts-jest problems eventually I guess). All interesting changes are pulled out into those 3 PRs

**Comment by agilgur5** [comment]
Ah right, `ts-jest` is having issues because `jest`'s ESM support is still [experimental](https://jestjs.io/docs/ecmascript-modules), meaning some CJS is still needed. `ts-jest` now has [some support](https://kulshekhar.github.io/ts-jest/docs/guides/esm-support/) too now though, so might be able to get it to work that way.

**EDIT**: welp, that gives even more problems because some of the test code does not work in ESM mode (e.g. `__dirname`, `require` for `tslib`, and imports of CJS packages)

**Pull request closed without merging**